components/org.wso2.carbon.identity.application.authenticator.push/src/main/java/org/wso2/carbon/identity/application/authenticator/push/PushAuthenticator.java

```diff
@@ -0,0 +1,500 @@
+                    user.getTenantDomain());
+            request.getSession().setAttribute(PushAuthenticatorConstants.DEVICES_LIST, deviceList);
+            JSONObject object;
+            JSONArray array = new JSONArray();
+
+            for (Device device : deviceList) {
+                object = new JSONObject();
+                object.put(PushAuthenticatorConstants.DEVICE_ID, device.getDeviceId());
+                object.put(PushAuthenticatorConstants.DEVICE_NAME, device.getDeviceName());
+                object.put(PushAuthenticatorConstants.DEVICE_MODEL, device.getDeviceModel());
+                object.put(PushAuthenticatorConstants.LAST_TIME_USED, device.getLastUsedTime().toString());
+                array.add(object);
+            }
+
+            AuthDataDTO authDataDTO = new AuthDataDTOImpl();
+            context.setProperty(PushAuthenticatorConstants.CONTEXT_AUTH_DATA, authDataDTO);
+            AuthContextCache.getInstance().addToCacheByRequestId(new AuthContextcacheKey(sessionDataKey),
+                    new AuthContextCacheEntry(context));
+
+            if (deviceList.size() == 1) {
+                sendRequest(request, response, deviceList.get(0).getDeviceId(), sessionDataKey);
+            } else {
+
+                String string = JSONArray.toJSONString(array);
+                String devicesPage = getDevicesPage(context) + "?sessionDataKey=" + URLEncoder.encode(sessionDataKey,
+                        StandardCharsets.UTF_8.name()) + "&devices=" + URLEncoder.encode(string,
+                        StandardCharsets.UTF_8.name());
+                response.sendRedirect(devicesPage);
+            }
+
+        } catch (IOException e) {
+            String errorMessage = "Error occurred when trying to redirect to the registered devices page.";
+            throw new AuthenticationFailedException(errorMessage, e);
+        } catch (PushDeviceHandlerServerException e) {
+            String errorMessage = "Error occurred when trying to redirect to the registered devices page. "
+                    + "Devices were not found.";
+            throw new AuthenticationFailedException(errorMessage, e);
+        } catch (PushDeviceHandlerClientException e) {
+            String errorMessage = "Error when trying to redirect to registered devices page. "
+                    + "Authenticated user was not found.";
+            throw new AuthenticationFailedException(errorMessage, e);
+        } catch (SQLException e) {
+            String errorMessage = "SQL exception occurred when trying to get the device list.";
+            throw new AuthenticationFailedException(errorMessage, e);
+        } catch (UserStoreException e) {
+//            log.error("Redirecting to registered devices page failed. User was not found in userstore");
+            String errorMessage = "Error occurred when trying to get the authenticated user. "
+                + "User was not found in the userstore.";
```

Comment:
Do we know that this error is only causing when the user is not found in the userstore?